lua/lazygit/window.lua

```diff
@@ -80,6 +85,20 @@ local function open_floating_window()
   vim.cmd(cmd)
   cmd = [[autocmd WinLeave <buffer> silent! execute 'silent bdelete! %s']]
   vim.cmd(cmd:format(border_buffer))
+  vim.api.nvim_create_autocmd('VimResized', {
+    callback = function()
+      vim.defer_fn(function()
+        if not vim.api.nvim_win_is_valid(border_window) then
+          return
+        end
+        local new_width, new_height, new_row, new_col = get_window_pos()
+        api.nvim_win_set_config(border_window,
+          { width = new_width + 2, height = new_height + 2, relative = 'editor', row = new_row - 1, col = new_col - 1, })
+        api.nvim_win_set_config(win,
+          { width = new_width, height = new_height, relative = 'editor', row = new_row, col = new_col, })
+      end, 20)
+    end
+  })
```

Comment:
Can you elaborate on why `defer_fn` is needed? 

Would this not work?

```suggestion
  vim.api.nvim_create_autocmd('VimResized', {
    callback = function()
      if not vim.api.nvim_win_is_valid(border_window) then
        return
      end
      local new_width, new_height, new_row, new_col = get_window_pos()
      api.nvim_win_set_config(border_window,
        { width = new_width + 2, height = new_height + 2, relative = 'editor', row = new_row - 1, col = new_col - 1, })
      api.nvim_win_set_config(win,
        { width = new_width, height = new_height, relative = 'editor', row = new_row, col = new_col, })
    end
  })
```

Reply:
Unfortunately, if you try it out, the position/size after resizing would be messed up, I think it is some race between the event and when the new size detail gets updated

It causes the contents of the window to shift too much left

There is probably a better solution

Reply:
Without defering, this is how it looks - I'm pretty sure it looked worse for me yesterday, but maybe it was during playing around with parameters. Without the delay it's still a bit wonky - notice how LG expands outside the border window

https://github.com/kdheepak/lazygit.nvim/assets/167217/21ed9c41-97b1-4763-9047-04ba30b069f5